crates/sign/src/lib.rs

```diff
@@ -45,15 +45,24 @@ fn sign_message(msg: &Message, sk: &SecretKey) -> Signature {
 /// with `verify_message`.
 pub fn verify_hash(hash: Hash, signature: &Signature) -> Result<(), secp256k1::Error> {
     let msg = Message::from_digest(hash);
-    verify_message(&msg, signature)
+    let res = recover_from_message(&msg, signature);
+    match res {
+        Ok(_) => Ok(()),
+        Err(e) => Err(e),
+    }
 }
```

Comment:
Honestly, I'm not sure we even need this.
I've refactored it to keep the same interface in case it's in use somewhere, but the `recover_hash` function below does the same thing, but returns the recovered pubkey instead of `()`.